src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventCustomList.java

```diff
@@ -30,19 +39,50 @@ public View getView(int position, @Nullable View convertView, @Nullable ViewGrou
             view = LayoutInflater.from(context).inflate(R.layout.activity_view_habit_event_custom_list, parent, false);
         }
 
+        HabitEventModel habitEvent = habitEvents.get(position);
+
+        /* Views */
+
+        /**
+         * Text View for habit date
+         */
         TextView habitDate = view.findViewById(R.id.date);
+
+        /**
+         * Image View for location icon
+         */
         ImageView locationIcon = view.findViewById(R.id.location_icon);
+
+        /**
+         * Image View for description icon
+         */
         ImageView descriptionIcon = view.findViewById(R.id.comment_icon);
+
+        /**
+         * Image View for Image icon
+         */
         ImageView ImageIcon = view.findViewById(R.id.image_icon);
 
-        // Todo: Change the date of past habit events to actual habit event data from Firestore
+        // Set the date when the habit event occured
         DateFormat df = new SimpleDateFormat("MMMM dd, YYYY");
-        habitDate.setText(df.format(new Date()));
+        habitDate.setText(df.format(habitEvent.getDate()));
 
-        // Todo: Change the icons' opacity according to the actual habit events' data from Firestore
-        locationIcon.setImageResource(R.drawable.ic_map_marker_positive);
-        descriptionIcon.setImageResource(R.drawable.ic_comment_negative);
-        ImageIcon.setImageResource(R.drawable.ic_image_positive);
+        // Set the location, comment, and image icons accordingly
+        if (isNull(habitEvent.getLocation())) {
```

Comment:
I set the negative icon as a default for every icon in the layout so I think it would be better if we just set positive programmatically

Reply:
Fixed!